### PR TITLE
[RDY] Tweak target marker position on overmap

### LINKED ISSUE #20992: Overmap double target marker
Bug from dividing integer uneven width or height by 2.
overmap_target_marker

### PR DESCRIPTION
**Pull request by AMurkin**
Tweak target marker position on overmap when target is offscreen.
When target.x is within range by width of onscreen map - marker_x equals target.x.
Same for marker_y.

EDIT: Fixes #20992

### CODE REVIEW

**Comment by AMurkin** [comment]
I've decide to fix bug from dividing integer uneven width or height by 2.
Same lines.

**Comment by AMurkin** [comment]
Now i can build! ) Found errors. ( Will fix.

**Comment by AMurkin** [comment]
Ready. Any thoughts?

File: src/overmap.cpp [hunk 1]
```diff
@@ -2324,38 +2324,42 @@ void overmap::draw(WINDOW *w, WINDOW *wbar, const tripoint &center,
...
     if (has_target && blink &&
-        (target.x < cursx - om_half_width ||
-         target.x >= cursx + om_half_width  ||
-         target.y < cursy - om_half_height ||
-         target.y >= cursy + om_half_height)) {
+        (target.x < offset_x ||
+         target.x >= offset_x + om_map_width ||
+         target.y < offset_y ||
+         target.y >= offset_y + om_map_height)) {
+        int marker_x = ( target.x >= offset_x ) ? ( ( target.x < offset_x + om_map_width  ) ? target.x - offset_x : om_map_width  - 1) : 0;
```
Comment (kevingranade) [inline src/overmap.cpp]: Nested ternary expressions make the code hard to read, please break this into a series of if statements, or multiple variables e.g. 'above_window', 'below_window', etc

**Comment by kevingranade** [comment]
Seems reasonable other than the ternary thing.

**Comment by AMurkin** [comment]
How about maximin?

**Pull request merged**